### ISSUE TITLE
Error parsing "SELECT AS STRUCT" on BigQuery

### ISSUE THREAD

**Issue by jml** [alanmcruickshank/sqlfluff#328]
```console
➜ sqlfluff version
0.3.4
```

Configuration:

```ini
[sqlfluff]
dialect = bigquery
templater = jinja

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L010]
capitalisation_policy = lower
```

SQL:

```sql
with

example as (
  select
    'foo' as a,
    1 as b

  union all

  select
    'bar' as a,
    2 as b
)

select as struct *
from example
```

Result: 

```console
➜ sqlfluff lint test.sql
== [test.sql] FAIL
L:  15 | P:  11 | L012 | Implicit aliasing of column not allowed. Use explicit
                       | `AS` clause.
L:  15 | P:  17 |  PRS | Found unparsable section: ' *'
```

The L012 warning is flat-out wrong. There's no aliasing at all. As a consequence, trying to fix that warning produces spurious SQL code

```console
➜ sqlfluff fix --rules L012 test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:  15 | P:  11 | L012 | Implicit aliasing of column not allowed. Use explicit
                       | `AS` clause.
==== fixing violations ====
1 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
== [test.sql] PASS
Done. Please check your files to confirm.

```

Resulting SQL:

```sql
with

example as (
  select
    'foo' as a,
    1 as b

  union all

  select
    'bar' as a,
    2 as b
)

select as AS struct *
from example
```

Note the line `select as AS struct *`, which has two errors:

1. It has two `as` keywords in a row
2. The added `AS` keyword is incorrectly upper-cased

Happily, the linter reports and fixes the first error:

```console
➜ sqlfluff lint test.sql
== [test.sql] FAIL
L:  15 | P:  11 | L010 | Inconsistent capitalisation of keywords.
L:  15 | P:  20 |  PRS | Found unparsable section: ' *'
➜ sqlfluff fix --rules L010 test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:  15 | P:  11 | L010 | Inconsistent capitalisation of keywords.
==== fixing violations ====
1 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
== [test.sql] PASS
Done. Please check your files to confirm.
```

Resulting in the following SQL:

```sql
with

example as (
  select
    'foo' as a,
    1 as b

  union all

  select
    'bar' as a,
    2 as b
)

select as as struct *
from example
```
